src/components/SiderMenu/SiderMenu.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, Suspense } from 'react'
+import { Layout } from 'antd'
+import classNames from 'classnames'
+import Link from 'umi/link'
+import styles from './index.less'
+import PageLoading from '../PageLoading'
+import { getDefaultCollapsedSubMenus } from './SiderMenuUtils'
+
+const BaseMenu = React.lazy(() => import('./BaseMenu'))
+const { Sider } = Layout
+
+function SiderMenu (props: any) {
+  const [openKeys, setOpenKeys] = useState(getDefaultCollapsedSubMenus(props))
+  const [pathname, setPathName] = useState(props.location.pathname)
+  const { menuData, logo, collapsed, onCollapse, fixSiderbar, theme } = props
+  const defaultProps = collapsed ? {} : { openKeys }
+  const siderClassName = classNames(styles.sider, {
+    [styles.fixSiderbar]: fixSiderbar,
+    [styles.light]: theme === 'light'
+  })
+
+  function isMainMenu (key) {
+    return menuData.some(item => {
+      if (key) {
+        return item.key === key || item.path === key
+      }
+      return false
+    })
+  }
```

Comment:
let's move it to the outer scope, and let's use for loop - it will be way more efficient (we won't need to create the whole temporar array):

```
function isMainMenu (menu, key) {
  if (!key) return false
  for(let item of menu) {
    if (item.key === key || item.path === key) return true
  }
  return false
}
```

Reply:
- [ ] moved to outer scope
- [ ] changed function